roboflow/models/video.py

```diff
@@ -0,0 +1,228 @@
+
+            >>> project = rf.workspace().project("PROJECT_ID")
+
+            >>> model = project.version("1").model
+
+            >>> prediction = model.predict("video.mp4", fps=5, inference_type="object-detection"
+        """
+
+        url = urljoin(API_URL, "/video_upload_signed_url/?api_key=", self.__api_key)
+
+        if fps > 5:
+            raise Exception("FPS must be less than or equal to 5.")
+        
+        for model in additional_models:
+            if model not in SUPPORTED_ADDITIONAL_MODELS:
+                raise Exception(f"Model {model} is not supported for video inference.")
+            
+        if inference_type not in SUPPORTED_ROBOFLOW_MODELS:
+            raise Exception(f"Model {inference_type} is not supported for video inference.")
+
+        if not is_valid_video(video_path):
+            raise Exception("Video path is not valid")
+
+        payload = json.dumps(
+            {
+                "file_name": video_path,
+            }
+        )
+
+        headers = {"Content-Type": "application/json"}
+
+        response = requests.request("POST", url, headers=headers, data=payload)
+
+        signed_url = response.json()["signed_url"]
+
+        print("Uploaded video to signed url: " + signed_url)
+
+        url = urljoin(API_URL, "/videoinfer/?api_key=", self.__api_key)
+
+        models = [
+            {
+                "model_id": self.dataset_id,
+                "model_version": self.version,
+                "inference_type": self.inference_type,
+            }
+        ]
+
+        for model in additional_models:
+            models.append(SUPPORTED_ADDITIONAL_MODELS[model])
+
+        payload = json.dumps(
+            {
+                "input_url": signed_url,
+                "infer_fps": 5,
+                "models": models
+            }
+        )
+
+        response = requests.request("POST", url, headers=headers, data=payload)
+
+        job_id = response.json()["job_id"]
+
+        self.job_id = job_id
+
+        return job_id, signed_url
+
+    def poll_for_results(self, job_id: str = None) -> dict:
+        """
+        Polls the Roboflow API to check if video inference is complete.
+
+        Returns:
+            Inference results as a dict
+
+        Example:
+            >>> import roboflow
+
+            >>> rf = roboflow.Roboflow(api_key="")
+
+            >>> project = rf.workspace().project("PROJECT_ID")
+
+            >>> model = project.version("1").model
+
+            >>> prediction = model.predict("video.mp4")
+
+            >>> results = model.poll_for_results()
+        """
+
+        if job_id is None:
+            job_id = self.job_id
+
+        url = urljoin(
+            API_URL, "/videoinfer/?api_key=", self.__api_key, "&job_id=", self.job_id
+        )
+
+        response = requests.get(url, headers={"Content-Type": "application/json"})
+
+        data = response.json()
+
+        if data["success"] != 0 or data["status_info"] != "success":
```

Comment:
if int(data['status']) == 0:
    success
elif int(data['status']) == 1:
    still running
else:
    failed....will retry it if int(data['retries']) <= 3